x-pack/plugins/endpoint/public/embeddables/resolver/view/edge_line.tsx

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import styled from 'styled-components';
+import { useSelector } from 'react-redux';
+import { applyMatrix3 } from '../lib/vector2';
+import { Vector2 } from '../types';
+import * as selectors from '../store/selectors';
+
+export const EdgeLine = styled(
+  React.memo(
+    ({
+      className,
+      startPosition,
+      endPosition,
+    }: {
+      className?: string;
+      startPosition: Vector2;
+      endPosition: Vector2;
+    }) => {
+      const projectionMatrix = useSelector(selectors.projectionMatrix);
+      const [left, top] = applyMatrix3(startPosition, projectionMatrix);
+      const length = distance(startPosition[0], startPosition[1], endPosition[0], endPosition[1]);
+      const deltaX = endPosition[0] - startPosition[0];
+      const deltaY = endPosition[1] - startPosition[1];
+      const angle = -Math.atan2(deltaY, deltaX);
+      /**
+       * https://www.mathsisfun.com/algebra/distance-2-points.html
+       */
+      function distance(x1, y1, x2, y2) {
+        return Math.sqrt(Math.pow(x2 - x1, 2) + Math.pow(y2 - y1, 2));
+      }
+
+      const style = {
+        left: left + 'px',
+        top: top + 'px',
+        width: length + 'px',
+        transformOrigin: 'top left',
+        transform: `translateY(-50%) rotateZ(${angle}rad)`,
+      };
+      return <div className={className} style={style} />;
```

Comment:
:question: I think `<line x1="${startPosition[0]}" x2="${${startPosition[0]}" y1="${endPosition[0]}" y2="${endPosition[0]}">` would be easier to read and deal with. Forcing these things into HTML/CSS space seems like it makes them a little resistant to some of the things we could do potentially do later to differentiate and polish this product as it matures (e.g. using paint servers for the stroke of the line, filters, masks, maybe animating parts of the transform). @oatkiller I know you said you wanted to defer the conversation until we started "rendering" things - and it looks like that's what we're starting to do here, right?  I'm OK if we decide as a team to just use `<div>`s and `<span>`s with CSS applied, but I still feel like a system that is specifically built in the Web to handle positioning and rendering things in a Cartesian coordinate system (like we're trying to do) is a better fit here. As fun as it is to see `-Math.atan2` in code we use, I think a casual viewer of the code seeing "x1, x2, y1, y2..." could be a little more confident about how it's working and what they change.  I acknowledge the downsides are 1) people aren't as familiar with SVG, so maybe that would defeat the whole purpose? 2) IE11 doesn't do CSS transforms in SVG space, so we'd be stuck with it. 3) SVG has more cross-browser issues. But I think the advantage of SVG being vastly more powerful visually and it being easier to "express" the kinds of things we're trying to do in coordinates makes it a better fit for the long game of the Resolver.

Reply:
seems worth investigating but probably out of the scope of this PR (unless that decision would change the graph layout logic.)

If you're interested in using SVGs here, you could make some progress by proving that we can nest any embeddable we want inside of svg?

Reply:
in any case, we'd still have atan and what not. changing a div to a line isn't going to change the math of laying out the graph right?

Reply:
Using the `line` obviates the need for aTan and other math: You give it coordinates of where it should start and end and it lays itself out.

Reply:
Which is a case in the overall point I'm making: It's 1 small thing here, but these complexities seem likely to "pile up" as we do more and more along the path of trying to torture CSS flow-based layouts to meet the coordinate-based needs of Resolver.

Reply:
@oatkiller w.r.t. the question of embedding HTML-based DOM elements in SVG, ForeignObject ( demonstrated here: https://bl.ocks.org/Jverma/2385cb7794d18c51e3ab ) is reported with support across most UAs (with the notable exception of Android for some reason: https://caniuse.com/#feat=mdn-api_svgforeignobjectelement )